splice_spark/src/test/scala/com/splicemachine/spark/splicemachine/TestContext.scala

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2012 - 2017 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.splicemachine.spark.splicemachine
+
+import java.math.BigDecimal
+import java.sql.{Time, Timestamp}
+import java.util.Date
+import org.apache.spark.sql.execution.datasources.jdbc.{JDBCOptions, JdbcUtils}
+import org.apache.spark.{SparkConf, SparkContext}
+import org.scalatest.{BeforeAndAfterAll, Suite}
+
+trait TestContext extends BeforeAndAfterAll { self: Suite =>
+
+  var sc: SparkContext = null
+  var splicemachineContext: SplicemachineContext = null
+  val table = "test"
+  val externalTable = "testExternal"
+  val schema = "TestContext"
+  val internalTN = schema+"."+table
+  val externalTN = schema+"."+externalTable
+
+  val appID = new Date().toString + math.floor(math.random * 10E4).toLong.toString
+  val defaultJDBCURL = "jdbc:splice://localhost:1527/splicedb;create=true;user=splice;password=admin"
+  val allTypesCreateStringWithPrimaryKey = "(" +
+    "c1_boolean boolean, " +
+    "c2_char char(5), " +
+    "c3_date date, " +
+    "c4_decimal numeric(15,2), " +
+    "c5_double double, " +
+    "c6_int int, " +
+    "c7_bigint bigint, " +
+    "c8_float float, " +
+    "c9_smallint smallint, " +
+    "c10_time time, " +
+    "c11_timestamp timestamp, " +
+    "c12_varchar varchar(56), " +
+    "primary key (c6_int, c7_bigint)" +
+     ")"
+  val allTypesCreateStringWithoutPrimaryKey = "(" +
+    "c1_boolean boolean, " +
+    "c2_char char(5), " +
+    "c3_date date, " +
+    "c4_decimal numeric(15,2), " +
+    "c5_double double, " +
+    "c6_int int, " +
+    "c7_bigint bigint, " +
+    "c8_float float, " +
+    "c9_smallint smallint, " +
+    "c10_time time, " +
+    "c11_timestamp timestamp, " +
+    "c12_varchar varchar(56)" +
+    ")"
+
+  val primaryKeys = Seq("c6_int","c7_bigint")
+
+  val allTypesInsertString = "(" +
+    "c1_boolean, " +
+    "c2_char, " +
+    "c3_date, " +
+    "c4_decimal, " +
+    "c5_double, " +
+    "c6_int, " +
+    "c7_bigint, " +
+    "c8_float, " +
+    "c9_smallint, " +
+    "c10_time, " +
+    "c11_timestamp, " +
+    "c12_varchar " +
```

Comment:
Any reason to avoid BLOB/CLOB types, maybe you could add separate IT for it if it is inconvenient here.